src/core/model/XojPage.h

```diff
@@ -56,17 +56,16 @@ class XojPage: public PageHandler {
 
     size_t getPdfPageNr() const;
 
-    bool isAnnotated();
+    bool isAnnotated() const;
 
     void setBackgroundColor(Color color);
     Color getBackgroundColor() const;
 
     std::vector<Layer*>* getLayers();
-    size_t getLayerCount();
-    int getSelectedLayerId();
-    void setSelectedLayerId(int id);
-    static bool isLayerVisible(Layer* layer);
-    bool isLayerVisible(int layerId);
+    size_t getLayerCount() const;
+    size_t getSelectedLayerId();
+    void setSelectedLayerId(size_t id);
```

Comment:
We should typedef the layer ids to avoid confusion, e.g.

```
    void setSelectedLayerId(Layer::Id id);
```

Reply:
We should have a clear scheme for this kind of things (maybe add it to the coding convention page). If we adopt `Layer::Index`, then the `Layer::ElementIndex` member type should probably be moved to `Element::Index`.

Reply:
> If we adopt Layer::Index, then the Layer::ElementIndex member type should probably be moved to Element::Index

Exactly

Reply:
Done!